src/main/scala/goatrodeo/omnibor/Builder.scala

```diff
@@ -195,27 +197,101 @@ object Builder {
       }
       t.join()
     }
-    println(f"Finished processing ${runningCnt.get()} at ${Duration
+    logger.info(f"Finished processing ${runningCnt.get()} at ${Duration
         .between(start, Instant.now())}")
 
     purlOut.flush()
     purlOut.close()
 
     val ret = storage match {
       case lf: (ListFileNames & Storage) if !dead_? =>
+        computeAndAddMerkleTrees(lf)
         writeGoatRodeoFiles(lf)
-      case _ => println("Didn't write"); None
+      case _ => logger.error("Didn't write"); None
     }
 
-    println(f"Total build time ${Duration.between(totalStart, Instant.now())}")
+    logger.info(
+      f"Total build time ${Duration.between(totalStart, Instant.now())}"
+    )
 
     ret
   }
 
+  /** Finds all the Items that "Contain" other items and generate a Merkle Tree
+    * alias
+    *
+    * @param data
+    *   the items
+    */
+  def computeAndAddMerkleTrees(data: ListFileNames & Storage): Unit = {
+    val keys = data.keys()
+
+    logger.info(f"Computing merkle trees for ${keys.length} items")
+    val merkleTrees = for {
+      k <- keys
+      item <- data.read(k)
+      if !item.connections.filter(a => EdgeType.isContainsDown(a._1)).isEmpty
+    } yield {
+      item.identifier -> GitOIDUtils.merkleTreeFromGitoids(
+        item.connections.toVector
+          .filter(a => EdgeType.isContainsDown(a._1))
+          .map(_._2)
+      )
+    }
+
+    logger.info(f"Computed ${merkleTrees.length} Merkle Trees")
+
+    // write the aliases
+    for { (itemId, tree) <- merkleTrees } {
+      // create the Merkle Tree Alias
+      data.write(
+        tree,
+        maybeAlias => {
+          val alias = maybeAlias.getOrElse(
+            Item(
+              identifier = tree,
+              reference = Item.noopLocationReference,
+              connections = TreeSet(),
+              bodyMimeType = None,
+              body = None
+            )
+          )
+          val toAdd = (EdgeType.aliasTo, itemId)
+          val updatedAlias =
+            if (alias.connections.contains(toAdd)) { alias }
+            else {
+              alias.copy(
+                connections = (alias.connections + toAdd)
+              )
+            }
+          updatedAlias
+        }
+      )
+      // and update the Item with the AliasFrom
+      data.write(
+        itemId,
+        originalItem =>
```

Comment:
nit: the `originalItem => originalItem match` clause is redundant and could be omitted

Reply:
Good call!

My bouncing between Scala and Rust results in me writing some "Rusty flavored Scala".